### ISSUE TITLE
map interaction

### ISSUE THREAD

**Issue by stevebutton** [mapbox/storytelling#20]
exploring this and testing in comparison to similar tools.
what may be a basic question: 
Is it possible to maintain the ability to interact with the map? 
In other words, at the moment I am locked to the specific views for each chapter - would like to be able to explore the map through normal mouse interactions. 
I go to a specific chapter/viewpoint, but I can still explore around that point.
Is this something in the config file or un-supported?

**Comment by jbranigan** [comment]
Hi @stevebutton , this template is designed to have a single type of interaction with the map: scrolling. This is intentional, as it lets you define how your readers experience the story (I imagine someone reading news on the subway). We are looking at ways to include some direct interaction, but it is more of a UX design question (e.g. how does the reader get back to scrolling after they switch to interactive mode?).

**Comment by stevebutton** [comment]
hi @jbranigan - thanks for getting back. Realize that this is what it is at the moment. 
I'm trying to explore the possibiliites of expanding along the lines of:
- scrolling through "destinations", as it does now
- on arriving at a given point, a button appears "explore this area", which switches off scrolling and allows for interaction.
- on activation the button switches to "continue journey" which when clicked re-activates scrolling. 
I realise this is not a support forum, but could you point me to what is activating/de-activating map interaction?

**Comment by jbranigan** [comment]
Hi @stevebutton, the interaction is currently blocked by a 100%-wide div overlaying the map container, so no touch events actually reach the map. 

For your use case, I'd consider maybe a click-through UI pattern (like previous/next). Still some design questions to work through, but it may simplify things for your team and be understandable to your users. Here's a great example of that type of approach (incidentally built by the folks behind scrollama.js): https://pudding.cool/2018/10/city_3d/

**Comment by laurenoldham1202** [comment]
@stevebutton I am new to Mapbox's storytelling functionality and I also want to enable general map interaction at the end of my story chapters, so I have put together a sort of hacky solution that may or may not suit your needs!

The basic idea is to use [jQuery](https://code.jquery.com/) to detach and re-append the story overlay that jbranigan mentioned above. So start by importing the jQuery CDN into your code.

Following the Glacier National Park example provided in the README, I first added an empty final chapter to the config.js file:
```
...
        {
            id: 'final-chapter',
            title: '',
            image: '',
            description: ``,
            location: {
                center: [-113.72917, 48.58938],
                zoom: 12.92,
                pitch: 39.50,
                bearing: 36.00
            },
            onChapterEnter: [],
            onChapterExit: []
        }
```
I then added a simple button on top of the map, set to invisible by default:
```
// CSS 
#return {
        z-index: 1000;
        position: absolute;
        top: 1rem;
        background: hotpink;
        font-size: 24pt;
        visibility: hidden;
}

// HTML
<div id="map"></div>
<button id="return">RETURN TO STORY</button>
<div id="story"></div>
```

Lastly, add this bit of code to the end of `scroller.onStepEnter()`, which basically says that once you enter the final chapter of your story, detach the story UI and reattach it on button click:
```
        if (response.index === config.chapters.length - 1) {
            story.detach();
            returnBtn.css('visibility', 'visible');
            returnBtn.on('click', () => {
                $('body').append(story);
                returnBtn.css('visibility', 'hidden');
            });
        }
```

The functionality isn't perfect as is - it currently still briefly displays the empty final chapter, etc. I'm sure you could make that UI invisible if you wanted, or instead use the final chapter's description to include an HTML button to close the story and explore the map. You can of course play around with the UI to suit your needs, but this should work for me!

**Comment by stevebutton** [comment]
Thanks Lauren - interesting approach. Will try this out and get back to you.

**Comment by mikelmaron** [comment]
I implemented another solution to this. In the chapter where map navigation is to wanted, set the callback to `enableFreetime`, and the callback on the preceding and following chapters to `disableFreetime`. Scrolling still controls moving through the story, hence the need to add and remove the NavigationControl.

This is the code added into `index.html`
<pre>
var nav = new mapboxgl.NavigationControl({visualizePitch: true});
function enableFreetime() {
  document.getElementById('map').style['z-index'] = 0;
  map.addControl(nav, 'top-right');
}
function disableFreetime() {
  document.getElementById('map').style['z-index'] = -5;
  if (map.hasControl(nav)) {
    map.removeControl(nav);
  }
}
</pre>

/cc @jbranigan

**Comment by davemaps** [comment]
@mikelmaron Where in the config.js chapters setup do we set the callback? First time working with this. Your solution seems elegant but I’m not sure how to implement it.

**Comment by mikelmaron** [comment]
Something like this @davemaps 

<pre>
     {
            id: 'last chapter',
            alignment: 'left',
            title: 'We are done',
            description: 'Wasn't that great?.',
            location: {
                center: [70.8426,42.2146],
                zoom: 12.66,
                pitch: 75,
                bearing: -12.8
            },
            callback: 'disableFreetime',
            onChapterEnter: [
            ],
            onChapterExit: [
            ]
        },
        {
            id: 'explore',
            alignment: 'left',
            title: 'Explore',
            image: '',
            description: '',
            location: {
              center: [65.12,48.39],
              zoom: 4.5,
              pitch: 4.00,
              bearing: 0.00
          },
            callback: 'enableFreetime',
            onChapterEnter: [
            ],
            onChapterExit: [
            ]
        }
</pre>

**Comment by davemaps** [comment]
@mikelmaron many thanks!

**Comment by chriswhong** [comment]
closing stale
